x.c

```diff
@@ -2047,14 +2060,8 @@ xdrawglyphfontspecs(const XftGlyphFontSpec *specs, Glyph base, int len, int x, i
 	}
 
 	if (base.mode & ATTR_REVERSE) {
-		if (bg == fg) {
-			bg = &dc.col[defaultfg];
-			fg = &dc.col[defaultbg];
-		} else {
-			temp = fg;
-			fg = bg;
-			bg = temp;
-		}
+		bg = &dc.col[defaultbg_reverse];
+		fg = &dc.col[defaultfg_reverse];
```

Comment:
i think this part (or smth else) is causing reversed text with custom bg to be displayed with same BG color as mouse-selected text

Reply:
oo thanks for the highlight -- I'm AFK till tomorrow but will give it a poke before I tag the 0.9 release

Reply:
oh you made a PR already -- redirecting